test/tests/functional/pbs_mom_dynamic_resource_ext.py

```diff
@@ -145,3 +175,177 @@ def test_t6(self):
 
         # The job should run
         self.server.expect(JOB, {'job_state': 'R'}, id=jid)
+
+    def test_t7(self):
+        """
+        Test for host level string resource
+        with string mom dynamic value
+        """
+
+        resc_name = "foo"
+        self.create_mom_resources(resc_name, "string", "h", "/bin/echo abc")
+
+        a = {'scheduling': 'True'}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+
+        # Submit a job that requests mom dynamic resource
+        attr = {"Resource_List." + resc_name: 'abc'}
+        j = Job(TEST_USER, attrs=attr)
+        jid = self.server.submit(j)
+
+        # The job should run
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid)
+
+    def test_t8(self):
+        """
+        host level consumable float resource with mom dynamic value float
+        """
+        resc_name = "foo"
+        self.create_mom_resources(resc_name, "float", "nh", "/bin/echo 3")
+
+        a = {'scheduling': 'True'}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+
+        # Submit a job that requests mom dynamic resource
+        attr = {"Resource_List." + resc_name: 4}
+        j = Job(TEST_USER, attrs=attr)
+        jid = self.server.submit(j)
+
+        # The job shouldn't run
+        c = "Can Never Run:"
+        c += " Insufficient amount of resource: foo (R: 4 A: 3 T: 3)"
+        self.server.expect(JOB, {'job_state': 'Q', 'comment': c},
+                           id=jid, attrop=PTL_AND)
+
+        # Submit a job that requests mom dynamic resource
+        attr = {"Resource_List." + resc_name: 2}
+        j = Job(TEST_USER, attrs=attr)
+        jid = self.server.submit(j)
+
+        # The job should run
+        self.server.expect(JOB, {'job_state': 'R',
+                           'Resource_List.'+resc_name: '2'},
+                           id=jid, attrop=PTL_AND)
+
+    def test_t9(self):
+        """
+        multiple host level resource long,float with all correct values.
+        Submit different jobs with different resource types
+        """
+        resc_name = ["foo", "foo1"]
+        resc_type = ["float", "long"]
+        script_body = ["/bin/echo 3", "/bin/echo 4"]
+        resc_flag = ["nh", "nh"]
+
+        self.create_mult_mom_resources(resc_name, resc_type,
+                                       resc_flag, script_body)
+
+        a = {'scheduling': 'True'}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+
+        # Submit a job that requests mom dynamic resource
+        attr = {"Resource_List." + resc_name[0]: 2,
+                "Resource_List." + resc_name[1]: 3}
+        j = Job(TEST_USER, attrs=attr)
+        jid = self.server.submit(j)
+
+        # The job should run
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid)
+
+        # Submit a job that requests no. mom resource
+        # greater than scrtipt value
```

Comment:
update the comment as "Submit a job that requests more value than available for both resources"